provider/keystore/keystore.go

```diff
@@ -88,6 +107,7 @@ func NewKeystore(d ds.Batching, opts ...Option) (Keystore, error) {
 		requests:   make(chan operation),
 		close:      make(chan struct{}),
 		done:       make(chan struct{}),
+		logger:     log.Logger(cfg.loggerName),
```

Comment:
Why have a configurable logger name?

If the logger was initialized with a fixed name,
```go
logger := log.Logger("keystore")
```
 then error message prefix would not be needed:
 `s.logger.Error("keystore:  some error")` -->  `s.logger.Error("some error")`